_posts/2023/2023-07-08-cobbler-inside-libvirt-2.md

```diff
@@ -0,0 +1,177 @@
+- RedHat:
+  - [RedHat Documentation - Configuring a network bridge](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/configuring_and_managing_networking/configuring-a-network-bridge_configuring-and-managing-networking)
+  - [RedHat Documentation - Virtual networking using bridges](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/configuring_and_managing_virtualization/configuring-virtual-machine-network-connections_configuring-and-managing-virtualization#virtual-networking-bridged-mode_types-of-virtual-machine-network-connections)
+  - [RedHat Sysadmin Blog](https://www.redhat.com/sysadmin/setup-network-bridge-VM)
+- Debian: [Debian Wiki](https://wiki.debian.org/BridgeNetworkConnections#Libvirt_and_bridging)
+- Vendor Neutral (the hard way): [Linuxconfig](https://linuxconfig.org/how-to-use-bridged-networking-with-libvirt-and-kvm)
+
+To read more about:
+
+- **NAT**: please head over to [Wikipedia](https://en.wikipedia.org/wiki/Network_address_translation).
+- **Network Briding**: please head over to [Wikipedia](https://en.wikipedia.org/wiki/Network_bridge)
+
+### Setup Cobbler VM & install the guest OS
+
+Download the openSUSE Tumbleweed ISO and move it to your desired libvirt storage pool (below I show the location of the 
+default storage pool):
+
+```shell
+wget https://download.opensuse.org/tumbleweed/iso/openSUSE-Tumbleweed-NET-x86_64-Current.iso
+mv openSUSE-Tumbleweed-NET-x86_64-Current.iso /var/lib/libvirt/images
+```
+
+Now we can start the setup of the guest. An example of a guide can be found
+[in the openSUSE Documentation](https://doc.opensuse.org/documentation/leap/virtualization/html/book-virtualization/cha-kvm-inst.html#sec-libvirt-inst-vmm). 
+
+During the setup two things need to be thought of:
+
+1. Give the VM a little more storage for the ISOs, 50GB is recommended. If you forget this you may not be able to 
+   download the ISO for the guest later on. Cobbler copies a lot from the ISO and as such be prepared that you need 
+   double the storage that the ISO takes up.
+2. Instead of the network `default` assign the network bridge so Cobbler receives and IP from your external DHCP server.
+
+### Install Cobbler on the Guest
+
+To make access to the host easier you could copy SSH keys to your guest from your host with the following command:
+
+```shell
+ssh-copy-id -i ~/.ssh/id_rsa <user>@<guest ip>
+```
+
+Passwordless login should now be possible with the following command:
+
+```shell
+ssh -i ~/.ssh/id_rsa <user>@<guest ip>
+```
+
+To install Cobbler on openSUSE Tumbleweed use the following command:
+
+```shell
+zypper in cobbler
+```
+
+You want to confirm the installation with `y` and then press enter.
+
+### Configure Cobbler
+
+Since in the early boot stage normally your files are transferred via the TFTP protocol you must enable it with the
```

Comment:
Wonky wording here in my view. Maybe consider this:
```suggestion
Since your files are usually transferred via the TFTP protocol during the early boot stage, you must enable it with the
```